### PR TITLE
Regenerate CI build docker images

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

Regenerate CI build docker images and move them to a new registry

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml [hunk 1]
```diff
@@ -10,22 +10,22 @@ jobs:
     - task: Docker@2
       displayName: login
       inputs:
-        containerRegistry: onnxruntimeregistry
+        containerRegistry: onnxruntimebuildcache
```
Comment (edgchen1) [inline tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml:13]: this update is ok in the short term, but can we move to using get_docker_image.py at some point so the images can be built and cached automatically?

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml:13]: Sure. That would be great! This PR is meant to solve the current problem, to get a clean state of onnxruntimeregistry.

Comment (edgchen1) [reply tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml:13]: ok, that sounds good. just fyi, we shouldn't rely on images persisting in onnxruntimebuildcache after i enable the automatic cache clean up (#5843).

**Review by edgchen1 — APPROVED** [review]
:shipit:

**Pull request merged**